Panic on daemon startup (master)

Got this today (current build `2fe51ec`):

```
panic: sync: unlock of unlocked mutex

goroutine 1 [running]:
sync.(*Mutex).Unlock(0xc20817e010)
        /usr/local/go/src/sync/mutex.go:91 +0x9a
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).removeDevice(0xc20817e000, 0xc2081b8980, 0x19, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:1402 +0x2f9
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).deactivateDevice(0xc20817e000, 0xc2081ae780, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:1377 +0x393
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).verifyBaseDeviceUUID(0xc20817e000, 0xc2081ae780, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:709 +0x1c2
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).setupBaseImage(0xc20817e000, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:741 +0x1ee
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).initDevmapper(0xc20817e000, 0xc2081c6001, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:1246 +0x15b2
github.com/docker/docker/daemon/graphdriver/devmapper.NewDeviceSet(0xc2081b8860, 0x1c, 0x1, 0x0, 0x0, 0x0, 0x4a27bf, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:1879 +0x4e7
github.com/docker/docker/daemon/graphdriver/devmapper.Init(0xc2081b8860, 0x1c, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/devmapper/driver.go:44 +0x172
github.com/docker/docker/daemon/graphdriver.GetDriver(0xfe13b0, 0xc, 0xc2081b65e0, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/docker/docker/daemon/graphdriver/driver.go:102 +0x1a4
github.com/docker/docker/daemon/graphdriver.New(0xc2081b65e0, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/graphdriver/driver.go:126 +0xa9d
github.com/docker/docker/daemon.NewDaemon(0x1704800, 0xc208038ac8, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/daemon.go:606 +0x7bf
main.mainDaemon()
        /go/src/github.com/docker/docker/docker/daemon.go:150 +0x9ef
main.main()
        /go/src/github.com/docker/docker/docker/docker.go:99 +0x6ef

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 12 [chan receive]:
github.com/docker/docker/pkg/listenbuffer.(*defaultListener).Accept(0xc2081b82e0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/pkg/listenbuffer/buffer.go:73 +0xb6
net/http.(*Server).Serve(0xc2081ae0c0, 0x7ff178975f38, 0xc2081b82e0, 0x0, 0x0)
        /usr/local/go/src/net/http/server.go:1728 +0x92
github.com/docker/docker/api/server.(*HttpServer).Serve(0xc2081b8340, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/server.go:125 +0x4d
github.com/docker/docker/api/server.func·002(0x7ff178975f98, 0xc2081b8340)
        /go/src/github.com/docker/docker/api/server/server.go:101 +0x40
created by github.com/docker/docker/api/server.(*Server).ServeApi
        /go/src/github.com/docker/docker/api/server/server.go:105 +0x6ac

goroutine 10 [chan receive]:
github.com/docker/docker/api/server.(*Server).ServeApi(0xc20805ad40, 0xc208078f50, 0x1, 0x1, 0x0, 0x0)
        /go/src/github.com/docker/docker/api/server/server.go:110 +0x74f
main.func·003()
        /go/src/github.com/docker/docker/docker/daemon.go:136 +0x5f
created by main.mainDaemon
        /go/src/github.com/docker/docker/docker/daemon.go:142 +0x877

goroutine 11 [chan receive]:
github.com/docker/docker/daemon.func·021()
        /go/src/github.com/docker/docker/daemon/debugtrap_unix.go:17 +0x5c
created by github.com/docker/docker/daemon.setupDumpStackTrap
        /go/src/github.com/docker/docker/daemon/debugtrap_unix.go:20 +0x18e
```

--- a/daemon/graphdriver/devmapper/deviceset.go
+++ b/daemon/graphdriver/devmapper/deviceset.go
@@ -691,6 +691,9 @@ func getDeviceUUID(device string) (string, error) {
 }
 
 func (devices *DeviceSet) verifyBaseDeviceUUID(baseInfo *DevInfo) error {
+	devices.Lock()
+	defer devices.Unlock()
+
 	if err := devices.activateDeviceIfNeeded(baseInfo); err != nil {
 		return err
 	}
@@ -710,6 +713,9 @@ func (devices *DeviceSet) verifyBaseDeviceUUID(baseInfo *DevInfo) error {
 }
 
 func (devices *DeviceSet) saveBaseDeviceUUID(baseInfo *DevInfo) error {
+	devices.Lock()
+	defer devices.Unlock()
+
 	if err := devices.activateDeviceIfNeeded(baseInfo); err != nil {
 		return err
 	}

